File: Js/1dArray.js
var runningSum = function (nums) {
  let newArr = []
  let prev
  for (let i = 0; i < nums.length; i++) {
    if (i == 0) {
      prev = nums[i]
      newArr.push(prev)
    } else {
      prev = prev + nums[i]
      newArr.push(prev)
    }
  }
  return newArr
}

File: python/1dArray.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        nums = [1,2,3,4]
        newList=[]
        for i in range(len(nums)):
            if(i==0):
                prev=nums[i]
                newList.append(prev)
            else:
                prev=prev+nums[i]
                newList.append(prev)
        print(nums)
        return newList